command/operator_debug.go

```diff
@@ -758,30 +812,41 @@ func (c *OperatorDebugCommand) collectVault(dir, vault string) error {
 
 // writeBytes writes a file to the archive, recording it in the manifest
 func (c *OperatorDebugCommand) writeBytes(dir, file string, data []byte) error {
-	relativePath := filepath.Join(dir, file)
+	// Replace invalid characters in filename
+	filename := helper.CleanFilename(file, "_")
+
+	relativePath := filepath.Join(dir, filename)
 	c.manifest = append(c.manifest, relativePath)
 	dirPath := filepath.Join(c.collectDir, dir)
-	filePath := filepath.Join(dirPath, file)
+	filePath := filepath.Join(dirPath, filename)
 
 	// Ensure parent directories exist
 	err := os.MkdirAll(dirPath, os.ModePerm)
 	if err != nil {
-		// Display error immediately -- may not see this if files aren't written
 		c.Ui.Error(fmt.Sprintf("failed to create parent directories of \"%s\": %s", dirPath, err.Error()))
 		return err
 	}
 
+	// Ensure filename doesn't escape the sandbox of the capture directory
+	escapes := helper.PathEscapesSandbox(c.collectDir, filePath)
+	if escapes {
+		return fmt.Errorf("file path escapes capture directory")
```

Comment:
@davemay99 with the change in db0d3ee you can keep the `c.Ui.Error` in this `writeBytes` method, but that means this error gets silently dropped if you don't also include it here. So you'll want to either move the `c.Ui.Error` calls into the caller (which is what I originally recommended) or add a `c.Ui.Error` call here.

Reply:
Thanks @tgross!  I had to rerun CircleCI to get an unrelated test to pass, but I think 794dd62 completes the requested changes.